addons/website_sale/views/templates.xml

```diff
@@ -3384,10 +3390,13 @@
     <!-- Product page images -->
     <template id="website_sale.shop_product_images" name="Shop Product Images">
         <t t-set="product_images" t-value="product_variant._get_images() if product_variant else product._get_images()"/>
-        <t t-set="ribbon" t-value="product_variant.sudo().variant_ribbon_id or product.sudo().website_ribbon_id"/>
+        <t
+            t-set="ribbon"
+            t-value="product_variant._get_ribbon(combination_info)"
+        />
         <t t-set="bg_color" t-value="ribbon['bg_color'] or ''"/>
         <t t-set="text_color" t-value="ribbon['text_color']"/>
-        <t t-set="bg_class" t-value="ribbon._get_position_class()"/>
+        <t t-set="bg_class" t-value="ribbon._get_css_classes()"/>
```

Comment:
Is this needed? In the `website_sale.shop_product_*` templates I couldn't find any reference to `bg_class`